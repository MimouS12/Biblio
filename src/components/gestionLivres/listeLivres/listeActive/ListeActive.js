import React, { useState, useEffect  } from "react"
// useCallback,
import Livre from "../../livre/Livre"
import "./listeActive.css"
import BookForm from "../../formBook/BookForm"


function ListeActive({booksActive,deleteBook,AjouterLivre,MiseAjourLivre}) {

  const [isVisible, setIsVisible] = useState(false)
  const[modeAdmin,setModeAdmin]=useState(false)
 

  useEffect(()=>{
    let user = localStorage.getItem('user')
    if(user ==="admin"){
        setModeAdmin(true)
    }}, [])



 

  const toggleVisibility = () => {
    setIsVisible(!isVisible)
  }

 
 
  return (
    
    <div className="adherents-list">
  
       {modeAdmin &&( 
            <div>
          <div>  <button className="Toggle" data-testid ="OnclickAjouter" onClick={toggleVisibility}>Ajouter livre</button>
            </div> 
           <div> {isVisible && (
                <BookForm AjouterLivre={AjouterLivre}/>
             )}</div>
           </div>
        )} 
      <h1 className="Titre1">Liste des livres </h1>

         {booksActive.length !==0 ?(
                <> 
      <div> <table className="entete-tab">
     <tbody>
      <tr >
        <th>Libelle</th>
        <th></th><th></th><th></th><th></th><th></th><th></th><th></th>

        <th>auteur</th>
        <th></th>
      </tr>
      </tbody>
      </table>
      </div>
      
     
    <div>
 
  
      {booksActive.map((book) => {

         return <Livre
                    key={book.id}
                     id={book.id}
                     libelle={book.libelle}
                     auteur={book.auteur}
                     EAN ={book.EAN}
                     edition={book.edition}
                     nbExemplaires ={book.nbExemplaires}
                     liste="active"
                    deleteBook={deleteBook}
                    MiseAjourLivre={MiseAjourLivre}
                   
                />
      })}
               
             
      
 
    </div>
    </>
              ):(
                <div > la liste des livres actives est vide </div>
              )}
  </div>
)
}

export default ListeActive

