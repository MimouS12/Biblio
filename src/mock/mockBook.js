export default [
    {
      id: "1",
      EAN:9782714411501,
      libelle:"Les oiseaux se cachent pour mourir", 
      auteur:"Colleen McCullough",
      edition:2010,
      nbExemplaires:21,
    },
    {
      id: "2",
      EAN:9782266308472,
    libelle:"Jamais sans ma fille", 
    auteur:"Betty Mahmoody",
    edition:2003,
    nbExemplaires:1,
    },
    {
        id: "2",
        EAN:9782266308472,
      libelle:"Jamais sans ma fille", 
      auteur:"Betty Mahmoody",
      edition:2003,
      nbExemplaires:1,
      },
      {
        id: "2",
        EAN:9782266308472,
      libelle:"Jamais sans ma fille", 
      auteur:"Betty Mahmoody",
      edition:2003,
      nbExemplaires:1,
      },
      {id :"3",
      EAN:9782352876113,
      libelle:"Jane Eyre", 
      auteur:"Charlotte BrontÃ«",
      edition:2015,
      nbExemplaires:10,
     
      }
      ,
      {id :"4",
      EAN:9782266107532,
      libelle:"ffffffffff", 
      auteur:"kkkkkkkkkkk",
      edition:2015,
      nbExemplaires:10,
     
      },
      {id :"5",
      EAN:9782253005018,
      libelle:"Lmmmmmmmmmmmmmmmm", 
      auteur:"rrrrrrrrr",
      edition:2017,
      nbExemplaires:7,
     
      },
      {id :"6",
      EAN:9782070360024,
      libelle:"llllllllllllll", 
      auteur:"oooooooooooo",
      edition:2001,
      nbExemplaires:11,
      
      },
      {id :"7",
      EAN:9782203135215,
      libelle:"ttttttttttttt", 
      auteur:"pppppppp",
      edition:2004,
      nbExemplaires:18,
     
      },
      {id :"8",
      EAN:9782070413119,
      libelle:"yyyyyyyyyyyy", 
      auteur:"eeeeeeeeeeeee",
      edition:2006,
      nbExemplaires:25,
      
      },
      {id :"9",
      EAN:9782253006220,
      libelle:"zzzzzzzzzz", 
      auteur:"pppppppppppppppp",
      edition:2007,
      nbExemplaires:9,
     
      }
      ,
      {id :"10",
      EAN:9782290231241,
      libelle:"hhhhhhhhhhh", 
      auteur:"dddddddddd",
      edition:2011,
      nbExemplaires:17,
     
      }
  ]