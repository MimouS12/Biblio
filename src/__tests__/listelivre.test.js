import React from "react"
import { render,fireEvent , act, waitForElementToBeRemoved} from "@testing-library/react"
//import TasksList from "./../components/TasksList/TasksList"
import mockBook from "../mock/mockBook"
import user from "@testing-library/user-event"
import { Router } from "react-router-dom"
import ListeActive from "../components/gestionLivres/listeLivres/listeActive/ListeActive"
import { createMemoryHistory } from "history"
//import Livre from "../components/gestionLivres/livre/Livre"
//import ListeLivres from "../components/gestionLivres/listeLivres/ListeLivres"
//import { createMemoryHistory } from "history"
/*import {
  updateBook as mockupdateBook,
  fetchBooks as mockFetchbooks,
 fetchbooksActive as mockfetchbooksActive
} from "../services/livres.service"*/
//jest.mock("../services/livres.service")

describe("test tasks list", () => {
  //hedha test ta3 snapdhot ki nbadlou ay 7aja fel page js ta3 component ta3na snapshot ywali failed mayet3adech 5tr howa ya3mllna copie mel component ta3na
  test("should render an array of tasks list", () => {
    const mockdeleteBook = jest.fn()
    const history = createMemoryHistory({ initialEntries: ["/"] })
    const BooksComponent = render(
      <Router history={history}>  <ListeActive deleteBook={mockdeleteBook} books={mockBook} /></Router>
    )
    expect(BooksComponent).toMatchSnapshot()
    // debug()
  }) 

 

test("should works without crashing", () => {
  //hedha test ta3 yjareb t5arjlou liste des livres mrigla
  const history = createMemoryHistory({ initialEntries: ["/"] })
  const { debug } = render(
    <Router history={history}>
    <ListeActive/> 
  </Router> 
  )
  // debug()
}) 
test('renders "no books" when there are an empty list of books', () => {
  //hedha test ki tabda liste fer4a par exemple nemchiw ncommentiw fi service les livres lkol lahn√© yet3ada test mta3na s7i7
  const history = createMemoryHistory({ initialEntries: ["/"] })
  const {debug,getByText} = render(<Router history={history}>
    <ListeActive/> 
  </Router>)
// debug()
 const listevide =getByText(/la liste des livres actives est vide/i)
  expect(getByText(/la liste des livres actives est vide/i)).toBeInTheDocument()
 // debug(listevide) 
})

/* test('renders contacts', () => {
  const fakeContacts = [{id: 1, name: 'Bob'}, {id: 2, name: 'Marcy'}]
  const {getAllByTestId} = render(<ContactList contacts={fakeContacts} />)
  const contactNames = getAllByTestId('contact-name').map(li => li.textContent)
  const fakeContactNames = fakeContacts.map(c => c.name)
  expect(contactNames).toEqual(fakeContactNames)*/

 
})