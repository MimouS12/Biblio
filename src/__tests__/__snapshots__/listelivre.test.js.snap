// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test tasks list should render an array of tasks list 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
        
      <div
        class="adherents-list"
      >
        <h1
          class="Titre1"
        >
          Liste des livres 
        </h1>
        <div
          class="search"
        >
          <input
            name="search"
            placeholder=" titre/nom auteur"
            type="search"
            value=""
          />
        </div>
        <div>
           
          <table
            class="entete-tab"
          >
            <tbody>
              <tr>
                <th>
                  Libelle
                </th>
                <th />
                <th />
                <th />
                <th />
                <th />
                <th />
                <th />
                <th>
                  auteur
                </th>
                <th />
              </tr>
            </tbody>
          </table>
        </div>
        <div>
          <div>
            <div
              class="adherenttable"
            >
              <div
                aria-label="libelle"
                class="libelle"
                data-testid="libelle"
              >
                Jamais sans ma fille
              </div>
              <div
                class="auteur"
                data-testid="auteur"
              >
                Betty Mahmoody
              </div>
              <div
                class="ButtonSection"
              >
                <div>
                  <button
                    class="button"
                    data-testid="Emprunter"
                    style="background: Goldenrod;"
                  >
                    Emprunter
                  </button>
                </div>
                <div>
                  <a
                    href="/memberPage/listelivres/2"
                  >
                     Détails
                  </a>
                </div>
              </div>
            </div>
          </div>
          <div>
            <div
              class="adherenttable"
            >
              <div
                aria-label="libelle"
                class="libelle"
                data-testid="libelle"
              >
                Jane Eyre
              </div>
              <div
                class="auteur"
                data-testid="auteur"
              >
                Charlotte Brontë
              </div>
              <div
                class="ButtonSection"
              >
                <div>
                  <button
                    class="button"
                    data-testid="Emprunter"
                    style="background: Goldenrod;"
                  >
                    Emprunter
                  </button>
                </div>
                <div>
                  <a
                    href="/memberPage/listelivres/3"
                  >
                     Détails
                  </a>
                </div>
              </div>
            </div>
          </div>
          <div>
            <div
              class="adherenttable"
            >
              <div
                aria-label="libelle"
                class="libelle"
                data-testid="libelle"
              >
                La cité de la joie
              </div>
              <div
                class="auteur"
                data-testid="auteur"
              >
                Dominique Lapierre
              </div>
              <div
                class="ButtonSection"
              >
                <div>
                  <button
                    class="button"
                    data-testid="Emprunter"
                    style="background: Goldenrod;"
                  >
                    Emprunter
                  </button>
                </div>
                <div>
                  <a
                    href="/memberPage/listelivres/4"
                  >
                     Détails
                  </a>
                </div>
              </div>
            </div>
          </div>
          <div>
            <div
              class="adherenttable"
            >
              <div
                aria-label="libelle"
                class="libelle"
                data-testid="libelle"
              >
                Le meilleur des Mondes
              </div>
              <div
                class="auteur"
                data-testid="auteur"
              >
                Aldous Huxley
              </div>
              <div
                class="ButtonSection"
              >
                <div>
                  <button
                    class="button"
                    data-testid="Emprunter"
                    style="background: Goldenrod;"
                  >
                    Emprunter
                  </button>
                </div>
                <div>
                  <a
                    href="/memberPage/listelivres/5"
                  >
                     Détails
                  </a>
                </div>
              </div>
            </div>
          </div>
          <div>
            <div
              class="adherenttable"
            >
              <div
                aria-label="libelle"
                class="libelle"
                data-testid="libelle"
              >
                L'étranger
              </div>
              <div
                class="auteur"
                data-testid="auteur"
              >
                Albert Camus
              </div>
              <div
                class="ButtonSection"
              >
                <div>
                  <button
                    class="button"
                    data-testid="Emprunter"
                    style="background: Goldenrod;"
                  >
                    Emprunter
                  </button>
                </div>
                <div>
                  <a
                    href="/memberPage/listelivres/6"
                  >
                     Détails
                  </a>
                </div>
              </div>
            </div>
          </div>
          <div>
            <div
              class="adherenttable"
            >
              <div
                aria-label="libelle"
                class="libelle"
                data-testid="libelle"
              >
                Les malheurs de Sophie
              </div>
              <div
                class="auteur"
                data-testid="auteur"
              >
                Comtesse de Ségur
              </div>
              <div
                class="ButtonSection"
              >
                <div>
                  <button
                    class="button"
                    data-testid="Emprunter"
                    style="background: Goldenrod;"
                  >
                    Emprunter
                  </button>
                </div>
                <div>
                  <a
                    href="/memberPage/listelivres/7"
                  >
                     Détails
                  </a>
                </div>
              </div>
            </div>
          </div>
          <div>
            <div
              class="adherenttable"
            >
              <div
                aria-label="libelle"
                class="libelle"
                data-testid="libelle"
              >
                Madame Bovary
              </div>
              <div
                class="auteur"
                data-testid="auteur"
              >
                Gustave Flaubert
              </div>
              <div
                class="ButtonSection"
              >
                <div>
                  <button
                    class="button"
                    data-testid="Emprunter"
                    style="background: Goldenrod;"
                  >
                    Emprunter
                  </button>
                </div>
                <div>
                  <a
                    href="/memberPage/listelivres/8"
                  >
                     Détails
                  </a>
                </div>
              </div>
            </div>
          </div>
          <div>
            <div
              class="adherenttable"
            >
              <div
                aria-label="libelle"
                class="libelle"
                data-testid="libelle"
              >
                La Mère
              </div>
              <div
                class="auteur"
                data-testid="auteur"
              >
                Pearl Buck
              </div>
              <div
                class="ButtonSection"
              >
                <div>
                  <button
                    class="button"
                    data-testid="Emprunter"
                    style="background: Goldenrod;"
                  >
                    Emprunter
                  </button>
                </div>
                <div>
                  <a
                    href="/memberPage/listelivres/9"
                  >
                     Détails
                  </a>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
      
    <div
      class="adherents-list"
    >
      <h1
        class="Titre1"
      >
        Liste des livres 
      </h1>
      <div
        class="search"
      >
        <input
          name="search"
          placeholder=" titre/nom auteur"
          type="search"
          value=""
        />
      </div>
      <div>
         
        <table
          class="entete-tab"
        >
          <tbody>
            <tr>
              <th>
                Libelle
              </th>
              <th />
              <th />
              <th />
              <th />
              <th />
              <th />
              <th />
              <th>
                auteur
              </th>
              <th />
            </tr>
          </tbody>
        </table>
      </div>
      <div>
        <div>
          <div
            class="adherenttable"
          >
            <div
              aria-label="libelle"
              class="libelle"
              data-testid="libelle"
            >
              Jamais sans ma fille
            </div>
            <div
              class="auteur"
              data-testid="auteur"
            >
              Betty Mahmoody
            </div>
            <div
              class="ButtonSection"
            >
              <div>
                <button
                  class="button"
                  data-testid="Emprunter"
                  style="background: Goldenrod;"
                >
                  Emprunter
                </button>
              </div>
              <div>
                <a
                  href="/memberPage/listelivres/2"
                >
                   Détails
                </a>
              </div>
            </div>
          </div>
        </div>
        <div>
          <div
            class="adherenttable"
          >
            <div
              aria-label="libelle"
              class="libelle"
              data-testid="libelle"
            >
              Jane Eyre
            </div>
            <div
              class="auteur"
              data-testid="auteur"
            >
              Charlotte Brontë
            </div>
            <div
              class="ButtonSection"
            >
              <div>
                <button
                  class="button"
                  data-testid="Emprunter"
                  style="background: Goldenrod;"
                >
                  Emprunter
                </button>
              </div>
              <div>
                <a
                  href="/memberPage/listelivres/3"
                >
                   Détails
                </a>
              </div>
            </div>
          </div>
        </div>
        <div>
          <div
            class="adherenttable"
          >
            <div
              aria-label="libelle"
              class="libelle"
              data-testid="libelle"
            >
              La cité de la joie
            </div>
            <div
              class="auteur"
              data-testid="auteur"
            >
              Dominique Lapierre
            </div>
            <div
              class="ButtonSection"
            >
              <div>
                <button
                  class="button"
                  data-testid="Emprunter"
                  style="background: Goldenrod;"
                >
                  Emprunter
                </button>
              </div>
              <div>
                <a
                  href="/memberPage/listelivres/4"
                >
                   Détails
                </a>
              </div>
            </div>
          </div>
        </div>
        <div>
          <div
            class="adherenttable"
          >
            <div
              aria-label="libelle"
              class="libelle"
              data-testid="libelle"
            >
              Le meilleur des Mondes
            </div>
            <div
              class="auteur"
              data-testid="auteur"
            >
              Aldous Huxley
            </div>
            <div
              class="ButtonSection"
            >
              <div>
                <button
                  class="button"
                  data-testid="Emprunter"
                  style="background: Goldenrod;"
                >
                  Emprunter
                </button>
              </div>
              <div>
                <a
                  href="/memberPage/listelivres/5"
                >
                   Détails
                </a>
              </div>
            </div>
          </div>
        </div>
        <div>
          <div
            class="adherenttable"
          >
            <div
              aria-label="libelle"
              class="libelle"
              data-testid="libelle"
            >
              L'étranger
            </div>
            <div
              class="auteur"
              data-testid="auteur"
            >
              Albert Camus
            </div>
            <div
              class="ButtonSection"
            >
              <div>
                <button
                  class="button"
                  data-testid="Emprunter"
                  style="background: Goldenrod;"
                >
                  Emprunter
                </button>
              </div>
              <div>
                <a
                  href="/memberPage/listelivres/6"
                >
                   Détails
                </a>
              </div>
            </div>
          </div>
        </div>
        <div>
          <div
            class="adherenttable"
          >
            <div
              aria-label="libelle"
              class="libelle"
              data-testid="libelle"
            >
              Les malheurs de Sophie
            </div>
            <div
              class="auteur"
              data-testid="auteur"
            >
              Comtesse de Ségur
            </div>
            <div
              class="ButtonSection"
            >
              <div>
                <button
                  class="button"
                  data-testid="Emprunter"
                  style="background: Goldenrod;"
                >
                  Emprunter
                </button>
              </div>
              <div>
                <a
                  href="/memberPage/listelivres/7"
                >
                   Détails
                </a>
              </div>
            </div>
          </div>
        </div>
        <div>
          <div
            class="adherenttable"
          >
            <div
              aria-label="libelle"
              class="libelle"
              data-testid="libelle"
            >
              Madame Bovary
            </div>
            <div
              class="auteur"
              data-testid="auteur"
            >
              Gustave Flaubert
            </div>
            <div
              class="ButtonSection"
            >
              <div>
                <button
                  class="button"
                  data-testid="Emprunter"
                  style="background: Goldenrod;"
                >
                  Emprunter
                </button>
              </div>
              <div>
                <a
                  href="/memberPage/listelivres/8"
                >
                   Détails
                </a>
              </div>
            </div>
          </div>
        </div>
        <div>
          <div
            class="adherenttable"
          >
            <div
              aria-label="libelle"
              class="libelle"
              data-testid="libelle"
            >
              La Mère
            </div>
            <div
              class="auteur"
              data-testid="auteur"
            >
              Pearl Buck
            </div>
            <div
              class="ButtonSection"
            >
              <div>
                <button
                  class="button"
                  data-testid="Emprunter"
                  style="background: Goldenrod;"
                >
                  Emprunter
                </button>
              </div>
              <div>
                <a
                  href="/memberPage/listelivres/9"
                >
                   Détails
                </a>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
